### PR TITLE
Display latest properties for related items

### PR DESCRIPTION
**Pull request by GyanP**
https://dev.sourcefabric.org/browse/SDESK-3681

### CODE REVIEW

File: apps/archive/archive.py [hunk 1]
```diff
@@ -303,7 +303,10 @@ def _handle_media_updates(self, updates, original, user):
             self._set_association_timestamps(item_obj, updates, new=False)
 
             stored_item.update(item_obj)
-            updates[ASSOCIATIONS][item_name] = stored_item
+            if 'test' in item_name:
```
Comment (ioanpocol) [inline apps/archive/archive.py]: On settings you can define multiple related items fields, I think 'test' is just the name of one of defined related items on your instance.
You have to get the list of related items fields and check them against the fields on associations.
The key on associations has field_name--number format

**Comment by ioanpocol** [comment]
Take a look at on_update and _handle_media_updates functions.

File: apps/archive/archive.py [hunk 2]
```diff
@@ -376,18 +376,18 @@ def replace(self, id, document, original):
...
-            raise SuperdeskApiError.forbiddenError("User does not have permissions to read the item.")
-
         # To fetch the related items json on the basis of ID stored in association for related items
         if item and 'associations' in item:
             for item_name, item_obj in item.get(ASSOCIATIONS).items():
-                if item_obj and 'test' in item_name:
+                if item_obj and self._is_related_content(item_name):
                     item_id = item_obj[config.ID_FIELD]
-                    stored_item = self.find_one(req=None, _id=item_id)
+                    stored_item = super().find_one(req=None, _id=item_id)
```
Comment (ioanpocol) [inline apps/archive/archive.py]: If the item is published a new item is created and this item should be retrieved here.
Also on every update or correction on a published item a new item is created and there is a flag last_published_version that helps you to find the last one.

File: apps/archive/archive.py [hunk 3]
```diff
@@ -887,6 +894,15 @@ def _add_desk_metadata(self, updates, original):
         """
         return get_resource_service('desks').apply_desk_metadata(updates, original)
 
+    def _is_related_content(self, item_name):
```
Comment (ioanpocol) [inline apps/archive/archive.py]: Can you retrieve the list only one time?

File: apps/archive/archive.py (hunk 2 above)
Comment (ioanpocol) [reply apps/archive/archive.py]: Sorry, my mistake, I think on update you don't have to retrieve the new created item, you still have to find the one with last_published_version set to true.

**Comment by GyanP** [comment]
@ioanpocol, I have made a small change to fix, when we try to open a published related item by clicking on edit icon it does not open.

**Comment by petrjasek** [comment]
@GyanP can you pls rebase it? it has some extra commits

**Comment by GyanP** [comment]
Rebased the PR

**Comment by ioanpocol** [comment]
@GyanP Please fix the failing tests

**Comment by ioanpocol** [comment]
@GyanP 
Is here anything remaining to be done?

**Pull request merged**